src/OIDCServiceProvider.php

```diff
@@ -53,8 +54,17 @@ final public function boot(): void
 
     private function getOIDCClient(): Client
     {
+        $this->autodiscovery();
+
         $config = collect(config('oidc'))
             ->put('redirect_uri', route('oidc.callback'));
         return new Client($config->all());
     }
+
+    private function autodiscovery(): void {
+        if (!is_null($providerURL = config('oidc.provider_url'))) {
+            $config = Http::get($providerURL)->json();
```

Comment:
Also, you should set the url to `PROVIDER_URL/.well-known/openid_configuration`, as stated by the official specs: https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationRequest